egg/core/interaction.py

```diff
@@ -131,7 +160,13 @@ def _check_cat(lst):
                 )
             return torch.cat(lst, dim=0)
 
-        assert interactions, "list must not be empty"
+        interactions = [
+            interact for interact in interactions if not interact.is_empty()
+        ]
+        if (
+            not interactions
+        ):  # handling the extreme case where all interactions have been filtered out
+            return Interaction.empty()
```

Comment:
I'm not 100% sure about this, does it look ok to you? It mostly came out as this [test](https://github.com/facebookresearch/EGG/blob/master/tests/test_toy_counting.py#L52) was failing without the check  @eugene-kharitonov